contracts/oracle/PCVOracle.sol

```diff
@@ -0,0 +1,388 @@
+        external
+        view
+        returns (bool)
+    {
+        return illiquidVenues.contains(illiquidVenue);
+    }
+
+    /// @notice check if a venue is in the list of illiquid venues
+    /// @param liquidVenue address to check
+    /// @return boolean whether or not the liquidVenue is in the illiquid venue list
+    function isLiquidVenue(address liquidVenue) external view returns (bool) {
+        return liquidVenues.contains(liquidVenue);
+    }
+
+    /// @notice check if a venue is in the list of liquid or illiquid venues
+    /// @param venue address to check
+    /// @return boolean whether or not the venue is part of the liquid or illiquid venue list
+    function isVenue(address venue) external view returns (bool) {
+        return liquidVenues.contains(venue) || illiquidVenues.contains(venue);
+    }
+
+    /// @notice get the total PCV balance by looping through the liquid and illiquid pcv deposits
+    /// @dev this function is meant to be used offchain, as it is pretty gas expensive.
+    function getTotalPcv()
+        external
+        view
+        returns (
+            uint256 liquidPcv,
+            uint256 illiquidPcv,
+            uint256 totalPcv
+        )
+    {
+        uint256 liquidVenueLength = liquidVenues.length();
+        uint256 illiquidVenueLength = illiquidVenues.length();
+
+        /// there will never be more than 100 total venues
+        /// so keep the math unchecked to save on gas
+        unchecked {
+            for (uint256 i = 0; i < liquidVenueLength; i++) {
+                address depositAddress = liquidVenues.at(i);
+                (Decimal.D256 memory oracleValue, bool oracleValid) = IOracle(
+                    venueToOracle[depositAddress]
+                ).read();
+                require(oracleValid, "PCVO: invalid oracle value");
+
+                liquidPcv +=
+                    (oracleValue.asUint256() *
+                        PCVDeposit(depositAddress).balance()) /
+                    1e18;
+            }
+
+            for (uint256 i = 0; i < illiquidVenueLength; i++) {
+                address depositAddress = illiquidVenues.at(i);
+                (Decimal.D256 memory oracleValue, bool oracleValid) = IOracle(
+                    venueToOracle[depositAddress]
+                ).read();
+                require(oracleValid, "PCVO: invalid oracle value");
+
+                illiquidPcv +=
+                    (oracleValue.asUint256() *
+                        PCVDeposit(depositAddress).balance()) /
+                    1e18;
+            }
+
+            totalPcv = liquidPcv + illiquidPcv;
+        }
+    }
+
+    /// ------------- PCV Deposit Only API -------------
+
+    /// @notice update the cumulative and last updated times
+    /// only callable by a liquid pcv deposit
+    /// this allows for lazy evaluation of the TWAPCV
+    /// @param pcvDelta the amount of PCV change in the venue
+    function updateLiquidBalance(int256 pcvDelta)
+        public
```

Comment:
add a modifier here that asserts the global reentrancy lock is true, because that's the only time this function should be able to be called.